src/crypto/algorithms/megolm.ts

```diff
@@ -605,12 +606,11 @@ class MegolmEncryption extends EncryptionAlgorithm {
             const userId = val.userId;
             const deviceInfo = val.deviceInfo;
             const deviceId = deviceInfo.deviceId;
-            deviceInfoByDeviceId.set(deviceId, deviceInfo);
 
-            if (!contentMap[userId]) {
-                contentMap[userId] = {};
+            if (userDeviceMapping[userId] == null) {
```

Comment:
relying on type coercion due to `==` is meh, it'd be `undefined` if anything but also can just be treated as falsey

Reply:
I personally prefer testing for nullish over falsey explicitly, also see #2124 where I got this from.

However, if this still isn't alright, ill just change it to falsey checking.

Reply:
I think conflating this otherwise useful change with a slightly controversial typing nit would be a waste, it'd just bikeshed it a little bit

Reply:
Alright, I'll change it to falsey then :+1: